tests/scalers/beanstalkd/beanstalkd_test.go

```diff
@@ -0,0 +1,267 @@
+	data, templates := getTemplateData()
+	t.Cleanup(func() {
+		DeleteKubernetesResources(t, testNamespace, data, templates)
+	})
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, beanstalkdDeploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 0 after a minute")
+
+	// test activation
+	testActivation(t, kc, data)
+
+	// test scaling in
+	testScaleOut(t, kc, data)
+
+	// scaling out
+	testScaleIn(t, kc, data)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:            testNamespace,
+			ScaledObjectName:         scaledObjectName,
+			DeploymentName:           deploymentName,
+			BeanstalkdDeploymentName: beanstalkdDeploymentName,
+			BeanstalkdTubeName:       beanstalkdTubeName,
+			BeanstalkdPutJobName:     beanstalkdPutJobName,
+			BeanstalkdPopJobName:     beanstalkdPopJobName,
+			JobCount:                 activationJobCount,
+		}, []Template{
+			{Name: "beanstalkdDeploymentTemplate", Config: beanstalkdDeploymentTemplate},
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+		}
+}
+
+// Adds five beanstalkd jobs to the default tube
+func addBeanstalkdJobs(t *testing.T, kc *kubernetes.Clientset, data *templateData) {
+	// run putJob
+	KubectlReplaceWithTemplate(t, data, "beanstalkdPutJobsTemplate", beanstalkdPutJobsTemplate)
+	require.True(t, WaitForJobSuccess(t, kc, beanstalkdPutJobName, testNamespace, 30, 2), "Job should run successfully")
+}
+
+// Removes five beanstalkd jobs from the default tube
+func removeBeanstalkdJobs(t *testing.T, kc *kubernetes.Clientset, data *templateData) {
+	// run putJob
+	KubectlReplaceWithTemplate(t, data, "beanstalkdPopJobsTemplate", beanstalkdPopJobsTemplate)
+	require.True(t, WaitForJobSuccess(t, kc, beanstalkdPopJobName, testNamespace, 30, 2), "Job should run successfully")
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation---")
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectActivationTemplate", scaledObjectActivationTemplate)
+
+	// Add 5 beanstalkd jobs
+	data.JobCount = 5
+	addBeanstalkdJobs(t, kc, &data)
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 30)
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scaling out ---")
+
+	// Add 100 beanstalkd jobs
+	data.JobCount = 100
+	addBeanstalkdJobs(t, kc, &data)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 3, 60, 1),
+		"replica count should be 3 after a minute")
+}
+
+func testScaleIn(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scaling in ---")
+
+	// Remove 80 beanstalkd jobs
+	data.JobCount = 80
+	removeBeanstalkdJobs(t, kc, &data)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 2, 3000, 1),
```

Comment:
I'd say that 3000 is a typo xD
```suggestion
	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 2, 60, 5),
```

Reply:
You are indeed correct this was a typo. Apologies, was obviously having a slow day! I've applied your suggestion and tested locally.